seed-job/index_reader.py

```diff
@@ -0,0 +1,369 @@
+
+    def __str__(self):
+        """
+        Returns the string representation of the project object
+        It returns the pipeline-name, which is constructed
+        based on parameters of the project indexed.
+        """
+        return self.pipeline_name
+
+    def replace_dot_slash_colon_(self, value):
+        """
+        Given a value with either dot slash or underscore,
+        replace each with hyphen
+        """
+        return value.replace("_", "-").replace("/", "-").replace(
+            ".", "-").replace(":", "-")
+
+    def process_depends_on(self, depends_on=None):
+        """
+        Process depends_on for given project based on entry index
+        and namespace
+        """
+        if not depends_on or depends_on == "null":
+            return None
+
+        if isinstance(depends_on, list):
+            return ",".join("{}-{}".format(
+                self.namespace,
+                self.replace_dot_slash_colon_(d))
+                for d in depends_on)
+        else:
+            return "{}-{}".format(
+                self.namespace,
+                self.replace_dot_slash_colon_(depends_on))
+
+    def process_desired_tag(self, desired_tag=None):
+        """
+        Process desired_tag for given project
+        """
+        if not desired_tag:
+            return "latest"
+        return desired_tag
+
+    def process_pre_build_script(self, prebuild_script=None):
+        """
+        Process prebuild_script for given project
+        """
+        if not prebuild_script:
+            return None
+        return prebuild_script
+
+    def process_pre_build_context(self, prebuild_context=None):
+        """
+        Process prebuild_context for given project
+        """
+        if not prebuild_context:
+            return None
+        return prebuild_context
+
+    def load_project_entry(self, entry):
+        """
+        Loads a container index entry in class objects
+        """
+        try:
+            self.app_id = self.replace_dot_slash_colon_(entry['app-id'])
+            self.job_id = self.replace_dot_slash_colon_(entry['job-id'])
+
+            self.git_url = entry['git-url']
+            self.git_path = entry['git-path']
+            self.git_branch = entry['git-branch']
+            self.target_file = entry['target-file']
+            self.build_context = entry.get('build-context', "./")
+            self.depends_on = self.process_depends_on(entry['depends-on'])
+            self.notify_email = entry['notify-email']
+            self.desired_tag = self.process_desired_tag(entry["desired-tag"])
+            self.pre_build_context = self.process_pre_build_context(
+                entry.get("prebuild-context", None))
+            self.pre_build_script = self.process_pre_build_script(
+                entry.get("prebuild-script", None))
+        except Exception as e:
+            print ("Error processing container index entry.")
+            raise(e)
```

Comment:
What's the intention behind raising an exception after catching it? What do we plan to do with it?

Reply:
to explicitly specify the operation in which exception is raised
```
"Error processing container index entry."
```
and also have traceback of what's failed.

An exception raised while processing container index entry should fail the seed-job build.

Reply:
:+1: didn't occur to me that failed `seed-job` can be used for alerting the admins.